### DISCUSSION TITLE
Ability to disable the Annotate and Download options

### DISCUSSION THREAD

**Discussion by anantiyer896** [oracle/opengrok#4299] [General]
It is easy for the developers to download an existing piece of code and copy it from another project, this causes a lot of copying of code to a senseless level, because of the annotation and download feature. 

If there were an option to disable the download and annotate options, it would be great. 

I wasn't able to find any solution to this problem as there is very little documentation on how to use and configure the opengrok system.

**Comment by vladak** [comment]
I think you are trying to address the issue (of copying pre-existing source code) at wrong spot. It is hard to imagine that OpenGrok instance is the only way for developers how to access source code, unless you lock their access really tightly. In general, I feel that this issue should be prevented by using appropriate tooling (to detect code similarities/duplications) and fostering desired behavior by educating/training.

While in general I am quite fond of tunables (maybe too much), this is not the right area, I think. The links in question are already dynamically displayed based on underlying feature availability.

That said, nothing is preventing you from forking and doing the changes in your fork.
